### PR TITLE
Added Kiota AuthProvider for SPFx solutions

### PR DESCRIPTION
**Pull request by luismanez**
This PR adds a new Kiota AuthProvider implementation to be used in SPFx solutions. The provider re-uses the existing Token provider provided by the same SharePoint Framework infrastructure. This provider can also be used for generated MS Graph Kiota clients.

### CODE REVIEW

File: packages/authentication/spfx/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[![npm version badge](https://img.shields.io/npm/v/@microsoft/kiota-authentication-spfx?color=blue)](https://www.npmjs.com/package/@microsoft/kiota-authentication-spfx)
+
+The Kiota Authentication SPFx Library is an implementation to authenticate HTTP requests in SPFx solutions. This Authentication library, re uses some infrastructure from the SPFx context to obtain a valid token for the required API.
+
+It also works for consuming MS Graph API, as Graph is just another Azure AD protected API.
```
Comment (baywet) [inline packages/authentication/spfx/README.md:5]:
```suggestion
It also works for consuming Microsoft Graph, as it is just another Azure Active Directory protected API.
```

File: packages/authentication/spfx/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        myCustomApiClient.teams.get().then(teams => {
+            // deal with returned data
+        })
+        .catch(e => {console.log(e)});
+      })
+      .catch(e => {console.log(e)});
+```
+
+If you have generated a Kiota client for MS Graph API, you can also use this Auth provider in a similar way:
```
Comment (baywet) [inline packages/authentication/spfx/README.md:41]:
```suggestion
If you have generated a Kiota client for Microsoft Graph, you can also use this Auth provider in a similar way:
```

File: packages/authentication/spfx/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+```ts
+    this.props.aadTokenProviderFactory.getTokenProvider()
+      .then((tokenProvider: AadTokenProvider): void => {
+
+        const authProvider =
+          new AzureAdSpfxAuthenticationProvider(
+            tokenProvider, 
+            "https://graph.microsoft.com");
```
Comment (baywet) [inline packages/authentication/spfx/README.md:50]: shouldn't this be just the hostname, or the .default permission scope?

File: packages/authentication/spfx/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,462 @@
+{
+    "name": "@microsoft/kiota-authentication-spfx",
```
Comment (baywet) [inline packages/authentication/spfx/package-lock.json:2]: we're using lerna here, which means the libs don't have a package lock (managed by lerna).
remove this file.
add any additional package this lib needs to the main package.json with yarn

File: packages/authentication/spfx/package.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        "OpenApi",
+        "Azure",
+        "Authentication",
+        "OAuth",
+        "SPFx",
+        "AzureAD",
+        "Graph",
+        "SharePoint"
+    ],
+    "author": "Luis Mañez",
```
Comment (baywet) [inline packages/authentication/spfx/package.json:35]: please use the same author as the other packages and add yourself as a contributor instead.

File: packages/authentication/spfx/tsconfig.base.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        "declaration": true,
+        "declarationMap": true,
+        "composite": true,
+        "paths": {
+            "@microsoft/kiota-abstractions": [
+                "../../abstractions"
+            ],
+        },
+
+        "skipLibCheck": true, // Otherwise, raises different issues with SPFx dependencies types (i.e: cannot find IInternalAadTokenProvider)
```
Comment (baywet) [inline packages/authentication/spfx/tsconfig.base.json:21]:
```suggestion
        "skipLibCheck": true
```
should be fine from the documentation, thanks for pointing out

File: packages/authentication/spfx/README.md (hunk 3 above)
Comment (luismanez) [reply packages/authentication/spfx/README.md:50]: yup, my bad! if you pass it, should be just the hostname, but by default is set to Microsoft Graph host, so in this case is not required.

File: packages/authentication/spfx/package.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    "contributors": [
+        {"name":"Luis Mañez","url":"https://github.com/luismanez"}
+    ],
+    "license": "MIT",
+    "bugs": {
+        "url": "https://github.com/microsoft/kiota-typescript/issues"
+    },
+    "homepage": "https://github.com/microsoft/kiota-typescript#readme",
+    "dependencies": {
+        "@azure/core-auth": "^1.3.2",
```
Comment (baywet) [inline packages/authentication/spfx/package.json:45]:
```suggestion
```
I don't think this one it needed for this package

**Review by baywet — CHANGES_REQUESTED** [review]

File: packages/authentication/spfx/package.json (hunk 7 above)
Comment (luismanez) [reply packages/authentication/spfx/package.json:45]: yeah, agreed, I think was there when I was testing the best way to mock the SPFx aadTokenProvider. thanks!

**Review by baywet — DISMISSED** [review]
Thanks for the great contribution and for addressing the feedback so quickly.

If you're looking to contribute more, a teams fx auth provider would be a great follow up to this and should not require any additional learning from you on the kiota side (low hanging fruit)

**Review by baywet — DISMISSED** [review]

**Review by baywet — DISMISSED** [review]

**Review by baywet — APPROVED** [review]

**Comment by sebastienlevert** [comment]
You can get some inspiration for the existing auth provider in MGT here : https://github.com/microsoftgraph/microsoft-graph-toolkit/tree/main/packages/providers/mgt-teamsfx-provider

This for this awesome contribution!

**Comment by luismanez** [comment]
Thanks for your help both. Will take a look to that teams fx thing and see what I can do :)

**Comment by baywet** [comment]
The package is up! https://www.npmjs.com/package/@microsoft/kiota-authentication-spfx
💪

**Pull request merged**